source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -0,0 +1,187 @@
+#include "AMGenericStepScanConfiguration.h"
+
+#include "ui/acquaman/AMGenericStepScanConfigurationView.h"
+#include "acquaman/AMGenericStepScanController.h"
+
+#include <math.h>
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(QObject *parent)
+	: AMStepScanConfiguration(parent)
+{
+	setName("Generic Scan");
+	setUserScanName("Generic Scan");
+}
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(const AMGenericStepScanConfiguration &original)
+	: AMStepScanConfiguration(original)
+{
+	setName(original.name());
+	setUserScanName(original.name());
+
+	computeTotalTime();
+
+	for (int i = 0, size = scanAxes_.count(); i < size; i++){
```

Comment:
I'm pretty sure that @dretrex has this one right. Using his coding paradigm, the call to scanAxes_.count() is done only once (to assign the value of the variable size).
Using your suggestion the call to scanAxes_.count() should be executed for each iteration of the loop.